structopt-derive/src/attrs.rs

```diff
@@ -59,10 +81,48 @@ impl ::std::str::FromStr for Parser {
     }
 }
 
+impl NameCasing {
+    fn translate(&self, input: &str) -> String {
+        match *self {
+            NameCasing::Camel => input.to_camel_case(),
+            NameCasing::Kebab => input.to_kebab_case(),
+            NameCasing::Mixed => input.to_mixed_case(),
+            NameCasing::ShoutySnake => input.to_shouty_snake_case(),
+            NameCasing::Snake => input.to_snake_case(),
+            NameCasing::Verbatim => String::from(input),
+        }
+    }
+}
+
+impl ::std::str::FromStr for NameCasing {
+    type Err = String;
+
+    fn from_str(name: &str) -> Result<Self, Self::Err> {
+        let name = name.to_camel_case().to_lowercase();
+
+        let case = match name.as_ref() {
+            "camel" | "camelcase" => NameCasing::Camel,
+            "kebab" | "kebabcase" => NameCasing::Kebab,
+            "mixed" | "mixedcase" => NameCasing::Mixed,
+            "shoutysnake" | "shoutysnakecase" => NameCasing::ShoutySnake,
+            "snake" | "snakecase" => NameCasing::Snake,
+            "verbatim" | "verbatimcase" => NameCasing::Verbatim,
+            _ => return Err(format!("unsupported casing {}", name)),
```

Comment:
could be great to support serde names: https://serde.rs/container-attrs.html

Reply:
The names defined by serde collide with the ones from heck (e.g. `camel case` in serde is `mixed case` in heck, and `camel case` in heck is `pascal case` in serde).  I would therefore need to drop the heck names.

Reply:
I think it's better to follow serde names as it is well known by the rust community. Are you OK with that?